src/Formats/FormatFactory.h

```diff
@@ -141,6 +141,7 @@ class FormatFactory final : private boost::noncopyable
         bool supports_parallel_formatting{false};
```

Comment:
I've noticed all of this is implemented differently than in AggregateFunctionFactory; see AggregateFunctionProperties.

We need something that will also be suitable for:
- system tables;
- autogenerated documentation.

Interesting, what approach is better.
We can unify it in a subsequent PR.

Reply:
Not sure what you mean. That the properties are grouped into a struct, and formats register the whole struct instead of a bunch of separate flags? Or something in parallel formatting or large blocks?